data-analytics/postgres-to-bigquery/src/main/python/postgres_to_bigquery_utils.py

```diff
@@ -0,0 +1,346 @@
+                              "WHEN is_nullable::boolean='f' THEN 'REQUIRED' " \
+                              "END as mode ," \
+                              "CASE " \
+                              "WHEN (data_type = 'text' OR data_type='varchar' " \
+                              "OR data_type ='character varying' OR data_type = 'unknown' " \
+                              "OR data_type='char' OR data_type='character') THEN 'STRING' " \
+                              "WHEN (data_type = 'integer' OR data_type='smallint' " \
+                              "OR data_type='bigint') THEN 'INTEGER' " \
+                              "WHEN (data_type = 'boolean') THEN 'BOOLEAN' " \
+                              "WHEN (data_type = 'real' OR data_type = 'double precision' " \
+                              "OR data_type='numeric' OR data_type='decimal') THEN 'FLOAT' " \
+                              "WHEN (data_type='date') THEN 'DATE' " \
+                              "WHEN (data_type='time' OR data_type='time without time zone' " \
+                              "OR data_type='time with time zone') THEN 'TIME' " \
+                              "WHEN (data_type='timestamp' OR data_type='timestamp without time zone' " \
+                              "OR data_type='timestamp with time zone') THEN 'TIMESTAMP' " \
+                              "END AS type " \
+                              "FROM information_schema.columns " \
+                              "WHERE table_schema = '{0}' and table_name ='{1}' " \
+                              "ORDER BY ordinal_position) table_json) TO STDOUT".format(schema_name, table_name)
+        get_json_data_sql_raw = "COPY (SELECT row_to_json(data_result) FROM (SELECT {} FROM {}.{}) data_result) TO STDOUT"
+        get_all_columns_sql = "SELECT column_name,data_type FROM information_schema.columns " \
+                              "WHERE table_schema = '{0}' AND table_name='{1}' ORDER BY ordinal_position" \
+            .format(schema_name, table_name)
+
+        connection = connection.get_conn()
+        cur = connection.cursor()
+        cur.execute(get_all_columns_sql)
+        columns_list = cur.fetchall()
+        columns_list_string = ""
+        for column in columns_list:
+            if column[1] != 'timestamp with time zone':
+                columns_list_string += ('"' + column[0] + '",')
+            else:
+                columns_list_string += ("EXTRACT(EPOCH FROM " + column[0] + ") AS " + column[0] + ",")
+
+        # Remove last comma
+        columns_list_string = columns_list_string[:-1]
+        get_json_data_sql = get_json_data_sql_raw.format(columns_list_string, schema_name, table_name)
+
+        schema_file_name = local_schema_dir + table_name + "_schema.json"
+        data_file_name = local_data_dir + table_name + "_data.json"
+
+        # Write Schema file
+        f = open(schema_file_name, 'wb')
+        f.write(bytes("[\n", "UTF-8"))
```

Comment:
Alternative proposal for this section (write to file, seek on file, write to file, run sed):

* Use an io.StringIO as the destination for cursor.copy_expert
* Use Python's string (str.replace or the regex module) manipulation to fix the doubled up backslashes
* Write the destination file once, and set the UTF-8 encoding on the output at write time